Initial commit

svitenium opened the pull request.

SimonVerhoeven committed:
Initial commit

Laoujin commented on src/belgianHolidays.ts:
```diff
@@ -0,0 +1,66 @@
+export class BelgianHolidays {
```
Geen class. Werk met functions.

Laoujin commented:
Ik zie geen testen?

Laoujin commented on LICENSE:
```diff
@@ -0,0 +1,13 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
```
Laat ons gewoon MIT nemen?
Weet eigenlijk niet wat het verschil tussen Apache2 en MIT is :)

Laoujin commented on CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+
+## 0.0.1 Initial draft
+- FEAT: added the functionality to fetch Belgian holidays for a given year
+- FEAT: added the functionality to determine whether a given date is a holiday
```
Delete deze maar

Laoujin commented on README.md:
```diff
@@ -1 +1,29 @@
+
+You can decide between fetching the full list of holidays with their names in Dutch, French & German, or checking for a specific date.
+
+` new BelgianHolidays().isHoliday(new Date(2023, 0, 1))` will result in `true`
```
new BelgianHolidays().isHoliday(new Date(2023, 0, 1, 15, 10, 3))

--> Gaat die ook TRUE geven?
--> Doordat je met die getTime werkt gaat dat mogelijk niet het geval zijn?

Of strip je het time part er ergens af?

svitenium replied:
ik heb de check aangepast

Laoujin commented on src/belgianHolidays.ts:
```diff
@@ -0,0 +1,66 @@
+        return new Date(year, M - 1, D);
+    }
+
+    private getEasterMonday(year: number) {
```
Doe full typing: `: Date` of `: bool`

svitenium replied:
ok

Laoujin commented on src/belgianHolidays.ts:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    public isHoliday(date: Date) {
+        return !!this.getHolidays(date.getFullYear()).find(item => {return item.date.getTime() === date.getTime()});
```
Er bestaat daarvoor een fn in JS: `some`

```
.some(hd => hd.date.getTime() === date.getTime())
```

Laoujin commented:
Okay looking good, deze mergen en we zijn er :)

svitenium committed:
Added missing test file

svitenium committed:
isHoliday now accepts any date, explicit typing, removed changelog

The API no longer expects midnight dates to be passed in, updated testcases,  removed the changelog for now and updated the license to MIT

svitenium commented:
> Ik zie geen testen?

Iets te ruime gitignore

svitenium committed:
Update README.md
